python/TestHarness/testers/CheckFiles.py

```diff
@@ -23,9 +23,13 @@ def getOutputFiles(self):
         return self.specs['check_files'] + self.specs['check_not_exists']
 
     def processResults(self, moose_dir, retcode, options, output):
-        output = FileTester.processResults(self, moose_dir, retcode, options, output)
+        reason = FileTester.processResults(self, moose_dir, retcode, options, output)
 
         specs = self.specs
+        if options.enable_recover and specs['skip_checks']:
```

Comment:
Ugh... I don't like this recover crap bleeding into the individual Testers. Can we handle this at the base class?